src/nvim/ex_docmd.c

```diff
@@ -6088,7 +6088,7 @@ static void ex_redrawstatus(exarg_T *eap)
   } else {
     status_redraw_curbuf();
   }
-  if (msg_scrolled && (State & MODE_CMDLINE)) {
+  if (msg_scrolled && !msg_use_msgsep() && (State & MODE_CMDLINE)) {
```

Comment:
looks like an ok workaround for now. in 0.9 `msg_use_msgsep()` will always be true, so the entire if statement can be yanked by then.